configure.ml

```diff
@@ -837,9 +837,16 @@ let strip =
 (** * md5sum command *)
 
 let md5sum =
-  if List.mem arch ["Darwin"; "FreeBSD"; "OpenBSD"]
-  then "md5 -q" else "md5sum"
-
+  let rec loop = function
+    | [] -> die "Don’t know how to compute MD5 checksums…"
+    | (p, ma, a) :: tl ->
+      if fst (tryrun p (ma @ a)) <> ""
+      then List.fold_left (Printf.sprintf "%s %s") p ma
+      else loop tl
+  in loop [
+    "md5sum", [], [ "--version" ];
+    "md5", ["-q"], [ "-s" ; "''" ];
+  ]
```

Comment:
Hi, `loop` seems of generic purpose, you could possibly move it next to `which` (of which is it somehow the dual). Don't know what @letouzey, author of the `configure.ml` script would say.

In the absence of other answers, good to me anyway.